lib/ladybug/energy_model/simulation_parameter.rb

```diff
@@ -0,0 +1,188 @@
+          #Gets or creates new SimulationControl object from the OpenStudio model
+          openstudio_simulation_control = @openstudio_model.getSimulationControl
+          if @hash[:simulation_control][:do_zone_sizing]
+            openstudio_simulation_control.setDoZoneSizingCalculation(@hash[:simulation_control][:do_zone_sizing])
+          else 
+            openstudio_simulation_control.setDoZoneSizingCalculation(@@schema[:definitions][:SimulationControl][:properties][:do_zone_sizing][:default])
+          end
+          if @hash[:simulation_control][:do_system_sizing]
+            openstudio_simulation_control.setDoSystemSizingCalculation(@hash[:simulation_control][:do_system_sizing])
+          else
+            openstudio_simulation_control.setDoSystemSizingCalculation(@@schema[:definitions][:SimulationControl][:properties][:do_system_sizing][:default])
+          end
+          if @hash[:simulation_control][:do_plant_sizing]
+            openstudio_simulation_control.setDoPlantSizingCalculation(@hash[:simulation_control][:do_plant_sizing])
+          else
+            openstudio_simulation_control.setDoPlantSizingCalculation(@@schema[:definitions][:SimulationControl][:properties][:do_plant_sizing][:default])
+          end
+          if @hash[:simulation_control][:run_for_run_periods]
+            openstudio_simulation_control.setRunSimulationforWeatherFileRunPeriods(@hash[:simulation_control][:run_for_run_periods])
+          else
+            openstudio_simulation_control.setRunSimulationforWeatherFileRunPeriods(@@schema[:definitions][:SimulationControl][:properties][:run_for_run_periods][:default])
+          end
+          if @hash[:simulation_control][:run_for_sizing_periods]
+            openstudio_simulation_control.setRunSimulationforSizingPeriods(@hash[:simulation_control][:run_for_sizing_periods])
+          else
+            openstudio_simulation_control.setRunSimulationforSizingPeriods(@@schema[:definitions][:SimulationControl][:properties][:run_for_sizing_periods][:default])
+          end
+        end
+        if @hash[:shadow_calculation]
+          openstudio_shadow_calculation = @openstudio_model.getShadowCalculation
+          if @hash[:shadow_calculation][:calculation_frequency]
+            openstudio_shadow_calculation.setCalculationFrequency(@hash[:shadow_calculation][:calculation_frequency])
+          else
+            openstudio_shadow_calculation.setCalculationFrequency(@@schema[:definitions][:ShadowCalculation][:properties][:calculation_frequency][:default])
+          end
+          if @hash[:shadow_calculation][:maximum_figures]
+            openstudio_shadow_calculation.setMaximumFiguresInShadowOverlapCalculations(@hash[:shadow_calculation][:maximum_figures])
+          else
+            openstudio_shadow_calculation.setMaximumFiguresInShadowOverlapCalculations(@@schema[:definitions][:ShadowCalculation][:properties][:maximum_figures][:default])
+          end
+        end
```

Comment:
Also, it looks like yo aren't assigning the `solar_distribution` property, which is something we included in the `shadow_calculation` schema here but I know EnergyPlus/OpenStudio stores this property on the `Building`. So I think you will have to get all Buildings in the OpenStudio Model and assign the `solar_distribution` property to them.

Reply:
@chriswmackey it is a method for the `SimulationControl` https://openstudio-sdk-documentation.s3.amazonaws.com/cpp/OpenStudio-1.8.0-doc/model/html/classopenstudio_1_1model_1_1_simulation_control.html. I can add it there.

Reply:
Ah, I didn't know that's how it is in OpenStudio SDK.  Clearly, all of us had the same reaction that it was out of place on the EnergyPlus Building object. I see that a bunch of the other fields that are on EnergyPlus Buildings are on the OpenStudio SimulationControl.  Thanks for the link and this is good to know for the future.